lib/ret_web/controllers/page_controller.ex

```diff
@@ -589,7 +605,44 @@ defmodule RetWeb.PageController do
     Plug.Static.call(conn, static_options)
   end
 
-  defp module_config(key) do
-    Application.get_env(:ret, __MODULE__)[key]
+  defp put_extra_response_headers(conn, key) do
+    (module_config(:"extra_#{key}_headers") || "")
+    |> String.split("|")
+    |> Enum.map(&String.trim/1)
+    |> Enum.map(&extra_header_to_tuple/1)
+    |> Enum.reduce(conn, fn {name, value}, conn -> conn |> put_resp_header(name, value) end)
+  end
+
+  defp extra_header_to_tuple(header) do
+    [name | rest] = header |> String.split(~r/:\s*/) |> Enum.map(&String.trim/1)
+
+    name =
+      name
+      |> String.downcase()
+      |> String.replace(~r/[^a-z0-9_-]/, "")
+
+    value = rest |> Enum.join(":")
```

Comment:
The split and join here could convert values from "foo: bar : baz" to "foo:bar:baz". Might be less surprising if you used the `parts` option on `split` to just extract the name.